### PR TITLE
Prevent errors when using string filters with empty values

### PR DESCRIPTION
**Pull request by licitdev**
## Issue

When filters are left empty, the raw value is `null`. Issue exists in the permission's field validation and in the field's condition.

image image

The following error is thrown when the filter is left empty. 

image

## Fix

Prevent the error by setting the Joi validation to `Joi.any().equal(true)` that returns the condition as false if the compareValue is `null` or `undefined`.

### CODE REVIEW

**Comment by rijkvanzanten** [comment]
(Lets add some tests 🙊)

File: packages/shared/src/utils/generate-joi.ts [hunk 1]
```diff
@@ -104,6 +104,12 @@ export function generateJoi(filter: FieldFilter, options?: JoiOptions): AnySchem
 		const getNumberSchema = () => (schema[key] ?? Joi.number()) as NumberSchema;
 		const getDateSchema = () => (schema[key] ?? Joi.date()) as DateSchema;
 
+		const preventEmptyString = (val: any) => {
+			if (typeof val === 'string') return val;
+			// Prevent errors in string functions from Joi and prevent coercing to an empty string
+			return val === null || val === undefined ? '==directus_missing_conditional_string==' : val.toString();
```
Comment (rijkvanzanten) [inline packages/shared/src/utils/generate-joi.ts:110]: Is there any way we can prevent having to rely on a hardcoded string like this? Even though it's probably unique enough, I'm also a little worried about falling back to an actual value that will still have the possibility to match 🤔

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]

File: packages/shared/src/utils/generate-joi.ts (hunk 1 above)
Comment (licitdev) [reply packages/shared/src/utils/generate-joi.ts:110]: @rijkvanzanten Can we force a `false` of the rule by setting the schema as `Joi.any().equal(true)` for string required filters but missing?

Comment (rijkvanzanten) [reply packages/shared/src/utils/generate-joi.ts:110]: There must be a way, but I don't know! Worth some research though 🙂

Comment (licitdev) [reply packages/shared/src/utils/generate-joi.ts]: Updated, no longer use a hardcoded string. The generated Joi schema using "string" type filters such as `_contains` or `_starts_with` with null values will always result in false.

Rationale: `null` or `undefined` is not equal to `true`, thus it'll be false.

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**